pkg/meta/meta.go

```diff
@@ -1004,6 +1006,57 @@ func (m *Meta) GetMetasByDBID(dbID int64) ([]structure.HashPair, error) {
 	return res, nil
 }
 
+// CheckSpecialAttributes checks if the special attributes are in the table info.
+// Make it same as hasSpecialAttributes.
+// Exported for testing.
+func CheckSpecialAttributes(str string) bool {
+	if strings.Contains(str, "\"tiflash_replica\":null") && strings.Contains(str, "\"policy_ref_info\":null") &&
```

Comment:
sometimes `str` is large, not sure it will run faster using this https://pkg.go.dev/index/suffixarray#New . Or we have prior knowledge about the order of these 4 substring so we can use strings/bytes.Index for one matching and use the index to skip the prefix when continue next matching. bytes package may be more suitable if strings package will copy and generate intermediate string objects.